src/range.rs

```diff
@@ -494,6 +570,93 @@ impl<T: Debug + Display + Clone + Eq + Ord> VersionSet for Range<T> {
     fn union(&self, other: &Self) -> Self {
         Range::union(self, other)
     }
+
+    fn is_disjoint(&self, other: &Self) -> bool {
+        let end_before_start = |end, start| match (end, start) {
```

Comment:
This is `!valid_segment` I think the file stays easier to read if we reduce the number of lookup table helpers, and document how they differ.